src/otp-input/otp-input.tsx

```diff
@@ -0,0 +1,206 @@
+                }
+            }
+        };
+
+    const handleKeyDown =
+        (index: number) => (event: React.KeyboardEvent<HTMLInputElement>) => {
+            const { key, code } = event;
+
+            const newOtpValues = [...otpValues];
+            if (
+                (code === "Backspace" || key === "Backspace") &&
+                newOtpValues[index] !== ""
+            ) {
+                newOtpValues[index] = "";
+            } else if (code === "Backspace" || key === "Backspace") {
+                newOtpValues[index - 1] = "";
+                inputRefs.current[index - 1]?.focus();
+            }
+
+            setOtpValues(newOtpValues);
+
+            if (onChange) {
+                onChange(newOtpValues.join(""));
+            }
+        };
+
+    const handlePaste = (event: ClipboardEvent): void => {
+        const pastedValue = event.clipboardData.getData("text");
+        if (pastedValue && validateUserInput(pastedValue, numOfInput)) {
+            setOtpValues(pastedValue.split(""));
+            if (onChange) {
+                onChange(pastedValue);
+            }
+        } else {
+            event.preventDefault();
+        }
+    };
+
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setOtpValues(new Array(numOfInput).fill(""));
+
+        if (!isWithinCooldown()) {
+            setLastCtaTimestamp(new Date());
+            setCountDown(cooldownDuration);
+        }
+
+        if (onClick) {
+            onClick(event);
+        }
+    };
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    const validateUserInput = (value: string, length = 1) =>
+        !value ? false : RegExp(`^[0-9]{${length}}$`).test(value);
+
+    const isWithinCooldown = (): boolean => {
+        const currentTime = Date.now();
+        const coolDownInMilliseconds = cooldownDuration * 1000;
+
+        return (
+            currentTime < lastCtaTimestamp.valueOf() + coolDownInMilliseconds
+        );
+    };
+
+    const beginCountdown = () => {
+        let timer = cooldownDuration;
+        const interval = setInterval(() => {
+            timer--;
+            setCountDown(timer);
+            if (timer === 0) {
+                clearInterval(interval);
+            }
+        }, 1000);
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderCTALabel = () => {
+        return otherCtaProps.children
+            ? `${otherCtaProps.children}`
+            : `Resend OTP${
+                  countDown
+                      ? ` in ${countDown} second${countDown > 1 ? "s" : ""}`
+                      : ""
+              }`;
+    };
+
+    const inputId = (index: number) =>
```

Comment:
This is a function but the name doesn't look like that. I think the generated id can be the same for the test id also. 
Suggestion is to create a function and keep it in `HELPER FUNCTIONS`

```suggestion
    const generateId = (index: number, prefix?: string) => {
        return prefix
            ? `${prefix}-otp-input-${index + 1}`
            : `otp-input-${index + 1}`;
    };

```